src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -128,30 +123,33 @@ def bulk_record(
             amount=cache_keys.size,
         )
 
-        cache_key_results = KeyResults()
-        cache_key_results.add_key_results(
-            [KeyResult.from_string(k, v) for k, v in cache_results.items() if v is not None],
+        cache_key_results = UseCaseKeyResults()
+        cache_key_results.add_use_case_key_results(
+            [UseCaseKeyResult.from_string(k, v) for k, v in cache_results.items() if v is not None],
             FetchType.CACHE_HIT,
         )
 
-        db_record_keys = cache_key_results.get_unmapped_keys(cache_keys)
+        db_record_keys = cache_key_results.get_unmapped_use_case_keys(cache_keys)
 
         if db_record_keys.size == 0:
             return cache_key_results
 
-        db_record_key_results = self.indexer.bulk_record(use_case_id, db_record_keys.mapping)
-        self.cache.set_many(
-            db_record_key_results.get_mapped_key_strings_to_ints(), use_case_id.value
+        db_record_key_results = self.indexer.bulk_record(
+            {
+                use_case_id: keyCollection.mapping
+                for use_case_id, keyCollection in db_record_keys.mapping.items()
+            }
         )
+        self.cache.set_many(db_record_key_results.get_mapped_strings_to_ints())
         return cache_key_results.merge(db_record_key_results)
 
-    def record(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
-        result = self.bulk_record(use_case_id=use_case_id, org_strings={org_id: {string}})
-        return result[org_id][string]
+    def record(self, use_case_id: UseCaseId, org_id: int, string: str) -> Optional[int]:
+        result = self.bulk_record(strings={use_case_id: {org_id: {string}}})
+        return result[use_case_id][org_id][string]
 
-    def resolve(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
-        key = f"{org_id}:{string}"
-        result = self.cache.get(key, use_case_id.value)
+    def resolve(self, use_case_id: UseCaseId, org_id: int, string: str) -> Optional[int]:
```

Comment:
We discussed this offline as well but let's make the changes to the signature of `resolve` and `reverse_resolve` in another PR. This will prevent us from having a big change with that and all the reference changes in a single PR.